### DISCUSSION TITLE
Restrict the scroll area in Canvas

### DISCUSSION THREAD

**Discussion by jayanthbondi** [xyflow/xyflow#2428] [General]
Hello React Flow Team,
I have been using your library for a good time and I'm really thankful to you.

I'm unable to find a way to restrict the scroll area in Canvas generated.
If I scroll the canvas using my mouse wheel, it keeps on scroll though the layout is not visible in the document.
I tried using translateExtent but couldn't find any solution.

It would be very helpful if you can suggest or share me any examples where you have handled this issue.
Also I recommend you to support a prop which can limit the scroll position based on the height of the layout.

Here's the sandbox link.
https://codesandbox.io/s/sandpack-project-forked-7skgrg?file=/App.js

**Comment by bcakmakoglu** [comment]
You guessed it right: Use `translateExtent`.
Here's your [sandbox](https://codesandbox.io/s/sandpack-project-forked-wkwc6x?file=/App.js) using that property.

**Reply by jayanthbondi** [reply]
Hi @bcakmakoglu ,
Thanks for the suggestion.
Since you have used [infinity,infinity], still I can scroll upto infinity to bottom.
Is there any way to limit the scroll only to the point where atleast one node is visible on the layout.

**Reply by bcakmakoglu** [reply]
Sure, there is a utility called `getRectOfNodes` that you can use to calculate a box around the currently visible nodes.
Then you can limit it to the width and height of that box. Add some padding if you need to.
Here's another [demo](https://codesandbox.io/s/sandpack-project-forked-ctyt0h?file=/Flow.js).

**Reply by jayanthbondi** [reply]
Thankyou @bcakmakoglu.
